optuna/multi_objective/visualization/_pareto_front.py

```diff
@@ -0,0 +1,124 @@
+import json
+from typing import List
+from typing import Optional
+
+from optuna._experimental import experimental
+from optuna.multi_objective.study import MultiObjectiveStudy
+from optuna.multi_objective.trial import FrozenMultiObjectiveTrial
+from optuna.visualization.plotly_imports import _imports
+
+if _imports.is_successful():
+    from optuna.visualization.plotly_imports import go
+
+
+@experimental("2.0.0")
+def plot_pareto_front(
+    study: MultiObjectiveStudy, names: Optional[List[str]] = None
+) -> "go.Figure":
+    """Plot the pareto front of a study.
+
+    Example:
+
+        The following code snippet shows how to plot the pareto front of a study.
+
+        .. testcode::
+
+            import optuna
+
+            def objective(trial):
+               x = trial.suggest_float("x", 0, 5)
+               y = trial.suggest_float("y", 0, 3)
+
+               v0 = (4 * x) ** 2 + (4 * y) ** 2
+               v1 = (x - 5) ** 2 + (y - 5) ** 2
+               return v0, v1
+
+            study = optuna.multi_objective.create_study(["minimize", "minimize"])
+            study.optimize(objective, n_trials=50)
+
+            optuna.multi_objective.visualization.plot_pareto_front(study)
+
+        .. raw:: html
+
+            <iframe src="../../_static/plot_pareto_front.html" width="100%" height="500px"
+            frameborder="0"></iframe>
+
+    Args:
+        study:
+            A :class:`~optuna.multi_objective.study.MultiObjectiveStudy` object whose trials
+            are plotted for their objective values.
+        names:
+            Objective name list used as the axis titles. If :obj:`None` is specified,
+            "Objective {objective_index}" is used instead.
+
+    Returns:
+        A :class:`plotly.graph_objs.Figure` object.
+
+    Raises:
+        :exc:`ValueError`:
+            If the number of objectives of ``study`` isn't 2 or 3.
+    """
+
+    _imports.check()
+
+    if study.n_objectives == 2:
+        return _get_pareto_front_2d(study, names)
+    elif study.n_objectives == 3:
+        return _get_pareto_front_3d(study, names)
+    else:
+        raise ValueError("`plot_pareto_front` function only supports 2 or 3 objective studies.")
+
+
+def _get_pareto_front_2d(study: MultiObjectiveStudy, names: Optional[List[str]]) -> "go.Figure":
+    if names is None:
+        names = ["Objective 0", "Objective 1"]
+    elif len(names) != 2:
+        raise ValueError("The length of `names` is supposed to be 2.")
+
+    trials = study.get_pareto_front_trials()
+
+    if len(trials) == 0:
+        raise ValueError("There must be one or more completed trials to plot a study.")
```

Comment:
Nit: I feel it suffices to append warnings to a log instead of raising an error.

Reply:
Your point sounds reasonable. I changed the behavior as you suggested: https://github.com/optuna/optuna/pull/1303/commits/42dd2deca23d7af8e59f88b0b0499ee2238acb63